### ISSUE TITLE
Can not push forwarding config to `tofino-model`

### ISSUE THREAD

**Issue by davidgengenbach** [stratum/stratum#485]
**Describe the bug**
Can not push P4 program to `tofino-model` setup described in https://github.com/stratum/stratum/blob/master/stratum/hal/bin/barefoot/README.run.md#running-stratum-on-tofino-model
Using `stratum-bfrt` instead of `stratum-bf` Docker tag.

**Environment**
- BF-SDE 9.3.0
- Ubuntu 18.04 LTS
- Stratum `master` branch

**To Reproduce**
1. Compile P4 program using TNA architecture 
 - `bf-p4c --target tofino --arch tna -o ./tna_simple_switch -g --verbose 3 --bf-rt-schema ./tna_simple_switch/bf-rt.json --p4runtime-files ./tna_simple_switch/p4info.txt  /home/dsp/dgengenbach/barefoot/bf-sde-9.3.0/pkgsrc/p4-examples/p4_16_programs/tna_simple_switch/tna_simple_switch.p4`
2. Pack forwarding config
  - https://github.com/stratum/stratum/blob/master/stratum/hal/bin/barefoot/README.pipeline.md
3. Build Stratum `tofino-model` image
  - https://github.com/stratum/stratum/blob/master/stratum/hal/bin/barefoot/README.run.md#running-stratum-on-tofino-model
4. Start `stratumproject/tofino-model:9.3.0` Docker tag
5. Start `stratumproject/stratum-bfrt:9.3.0` Docker tag
6. Wait for initialization to be complete
7. Push config using [`p4runtime-sh`](https://github.com/p4lang/p4runtime-shell)

**Expected behavior**
The forwarding config should be installed like on the hardware platform (Wedge100)

**Additional context**
The forwarding config can be pushed to `x86-64-accton-wedge100bf-32x-r0`.
The forwarding config program is taken from the BF-SDE examples, in this case: the `tna_simple_switch`.

<details>
  <summary>Logs</summary>
  
  ```
./execute.sh #
++ pwd -P
+ export SDE_TAR=/sapmnt/home/I539691/Projects/tmp/stratum-tofino_2020_12_19__10_51_12/tofino-model/bf_sde_9.3.0.tgz
+ SDE_TAR=/sapmnt/home/I539691/Projects/tmp/stratum-tofino_2020_12_19__10_51_12/tofino-model/bf_sde_9.3.0.tgz
+ main
+ run
+ docker kill tofino-model
tofino-model
+ docker run --detach --name tofino-model --rm -it --privileged --network=host stratumproject/tofino-model:9.3.0
cfe5499c83e0e55f0e4a3c84be5ea9f836f63682ae23a2f1c4ec0cf43112f76a
+ sleep 3
+ export SDE_VERSION=9.3.0
+ SDE_VERSION=9.3.0
+ export DOCKER_IMAGE=stratumproject/stratum-bfrt
+ DOCKER_IMAGE=stratumproject/stratum-bfrt
+ PLATFORM=barefoot-tofino-model
+ stratum/stratum/hal/bin/barefoot/docker/start-stratum-container.sh -bf_sim -bf_switchd_background=false -enable_onlp=false
++ uname -r
++ uname -r
+ docker run -it --rm --privileged -v /dev:/dev -v /sys:/sys -v /lib/modules/5.4.0-54-generic:/lib/modules/5.4.0-54-generic --env PLATFORM=barefoot-tofino-model --network host -v /var/log:/var/log/stratum stratumproject/stratum-bfrt:9.3.0 -bf_sim -bf_switchd_background=false -enable_onlp=false
Mounting hugepages...
Skipping kernel module installation.
bf_sysfs_fname /sys/class/bf/bf0/device/dev_add
Install dir: /usr (0x559d7bb38020)
bf_switchd: system services initialized
bf_switchd: loading conf_file /usr/share/stratum/tofino_skip_p4_no_bsp.conf...
bf_switchd: processing device configuration...
Configuration for dev_id 0
  Family        : Tofino
  pci_sysfs_str : /sys/devices/pci0000:00/0000:00:03.0/0000:05:00.0
  pci_domain    : 0
  pci_bus       : 5
  pci_fn        : 0
  pci_dev       : 0
  pci_int_mode  : 1
  sbus_master_fw: /usr/
  pcie_fw       : /usr/
  serdes_fw     : /usr/
  sds_fw_path   : /usr/
  microp_fw_path: 
bf_switchd: processing P4 configuration...
P4 profile for dev_id 0
  p4_name: dummy
    libpd: 
    libpdthrift: 
    context: 
    config: 
  diag: 
  accton diag: 
  board-port-map: /usr/share/port_map.json
  non_default_port_ppgs: 0
  SAI default initialize: 1 
board-port-map path: /usr/share/port_map.json
File /usr/share/port_map.json not found
Operational mode set to default: MODEL
bf_switchd: initializing dru_sim service
bf_switchd: library libdru_sim.so loaded
INFO: DRU sim MTI initialized successfully
dru_sim: client socket created
dru_sim: connected on port 8001
dru_sim: listen socket created
dru_sim: bind done on port 8002, listening...
dru_sim: waiting for incoming connections...
dru_sim: connection accepted on port 8002
dru_sim: DRU simulator running
bf_switchd: drivers initialized
Skipping P4 program load for dev_id 0
Setting core_pll_ctrl0=cd44cbfe

bf_switchd: dev_id 0 initialized

bf_switchd: initialized 1 devices
bf_switchd: Credo python server thread initialized..Skip p4 lib init
Skip mav diag lib init
bf_switchd: spawning cli server thread
credo_python_intf: listen socket created
credo_python_intf: bind done on port 9001, listening...
credo_python_intf: listening for incoming connections...
bf_switchd: spawning driver shell
bf_switchd: server started - listening on port 9999
I20201230 09:48:16.038480     1 main_bfrt.cc:78] switchd started successfully
Cannot read termcap database;
using dumb terminal settings.
W20201230 09:48:16.038554     1 credentials_manager.cc:45] Using insecure server credentials
I20201230 09:48:16.038666     1 hal.cc:128] Setting up HAL in COLDBOOT mode...
I20201230 09:48:16.038720     1 config_monitoring_service.cc:94] Pushing the saved chassis config read from /etc/stratum/barefoot-tofino-model/chassis_config.pb.txt...
bf-sde> I20201230 09:48:16.039862     1 bf_chassis_manager.cc:986] Port status notification callback registered successfully
I20201230 09:48:16.039978     1 bf_chassis_manager.cc:87] Adding port 1 in node 1 (SDK Port 0).
I20201230 09:48:16.040011     1 bf_chassis_manager.cc:116] Enabling port 1 in node 1 (SDK Port 0).
I20201230 09:48:16.040096     1 bf_chassis_manager.cc:87] Adding port 2 in node 1 (SDK Port 1).
I20201230 09:48:16.040119     1 bf_chassis_manager.cc:116] Enabling port 2 in node 1 (SDK Port 1).
I20201230 09:48:16.040180     1 bf_chassis_manager.cc:87] Adding port 3 in node 1 (SDK Port 2).
I20201230 09:48:16.040208     1 bf_chassis_manager.cc:116] Enabling port 3 in node 1 (SDK Port 2).
I20201230 09:48:16.040318     1 bf_chassis_manager.cc:87] Adding port 4 in node 1 (SDK Port 3).
I20201230 09:48:16.040349     1 bf_chassis_manager.cc:116] Enabling port 4 in node 1 (SDK Port 3).
I20201230 09:48:16.040428     1 bf_chassis_manager.cc:87] Adding port 5 in node 1 (SDK Port 4).
I20201230 09:48:16.040454     1 bf_chassis_manager.cc:116] Enabling port 5 in node 1 (SDK Port 4).
I20201230 09:48:16.040653     1 bf_chassis_manager.cc:87] Adding port 6 in node 1 (SDK Port 5).
I20201230 09:48:16.040681     1 bf_chassis_manager.cc:116] Enabling port 6 in node 1 (SDK Port 5).
I20201230 09:48:16.040889     1 bf_chassis_manager.cc:87] Adding port 7 in node 1 (SDK Port 6).
I20201230 09:48:16.040920     1 bf_chassis_manager.cc:116] Enabling port 7 in node 1 (SDK Port 6).
I20201230 09:48:16.041054     1 bf_chassis_manager.cc:87] Adding port 8 in node 1 (SDK Port 7).
I20201230 09:48:16.041085     1 bf_chassis_manager.cc:116] Enabling port 8 in node 1 (SDK Port 7).
I20201230 09:48:16.041244     1 bf_chassis_manager.cc:87] Adding port 9 in node 1 (SDK Port 8).
I20201230 09:48:16.041266     1 bf_chassis_manager.cc:116] Enabling port 9 in node 1 (SDK Port 8).
I20201230 09:48:16.041515     1 bf_chassis_manager.cc:87] Adding port 10 in node 1 (SDK Port 9).
I20201230 09:48:16.041554     1 bf_chassis_manager.cc:116] Enabling port 10 in node 1 (SDK Port 9).
I20201230 09:48:16.041693     1 bf_chassis_manager.cc:87] Adding port 11 in node 1 (SDK Port 10).
I20201230 09:48:16.041733     1 bf_chassis_manager.cc:116] Enabling port 11 in node 1 (SDK Port 10).
I20201230 09:48:16.041929     1 bf_chassis_manager.cc:87] Adding port 12 in node 1 (SDK Port 11).
I20201230 09:48:16.041958     1 bf_chassis_manager.cc:116] Enabling port 12 in node 1 (SDK Port 11).
I20201230 09:48:16.042189     1 bf_chassis_manager.cc:87] Adding port 13 in node 1 (SDK Port 12).
I20201230 09:48:16.042215     1 bf_chassis_manager.cc:116] Enabling port 13 in node 1 (SDK Port 12).
I20201230 09:48:16.042295     1 bf_chassis_manager.cc:87] Adding port 14 in node 1 (SDK Port 13).
I20201230 09:48:16.042320     1 bf_chassis_manager.cc:116] Enabling port 14 in node 1 (SDK Port 13).
I20201230 09:48:16.042407     1 bf_chassis_manager.cc:87] Adding port 15 in node 1 (SDK Port 14).
I20201230 09:48:16.042433     1 bf_chassis_manager.cc:116] Enabling port 15 in node 1 (SDK Port 14).
I20201230 09:48:16.042508     1 bf_chassis_manager.cc:87] Adding port 16 in node 1 (SDK Port 15).
I20201230 09:48:16.042536     1 bf_chassis_manager.cc:116] Enabling port 16 in node 1 (SDK Port 15).
I20201230 09:48:16.042721     1 bf_chassis_manager.cc:87] Adding port 17 in node 1 (SDK Port 16).
I20201230 09:48:16.042755     1 bf_chassis_manager.cc:116] Enabling port 17 in node 1 (SDK Port 16).
I20201230 09:48:16.042956     1 bfrt_switch.cc:60] Chassis config pushed successfully.
I20201230 09:48:16.045562     1 p4_service.cc:121] Pushing the saved forwarding pipeline configs read from /var/run/stratum/pipeline_cfg.pb.txt...
E20201230 09:48:16.045585     1 utils.cc:109] StratumErrorSpace::ERR_FILE_NOT_FOUND: /var/run/stratum/pipeline_cfg.pb.txt not found.
E20201230 09:48:16.045642     1 utils.cc:65] Return Error: ReadFileToString(filename, &text) failed with StratumErrorSpace::ERR_FILE_NOT_FOUND: /var/run/stratum/pipeline_cfg.pb.txt not found.
W20201230 09:48:16.045647     1 p4_service.cc:130] No saved forwarding pipeline config found at /var/run/stratum/pipeline_cfg.pb.txt. This is normal when the switch is just installed and no master controller is connected yet.
E20201230 09:48:16.046192     1 hal.cc:221] Stratum external facing services are listening to 0.0.0.0:28000, 0.0.0.0:9339, 0.0.0.0:9559, localhost:28000...
E20201230 09:48:16.059289     1 hal.cc:363] StratumErrorSpace::ERR_INTERNAL: Failed to check in with procmon: failed to connect to all addresses
E20201230 09:48:16.059321     1 hal.cc:231] Error when checking in with procmon: Failed to check in with procmon: failed to connect to all addresses.

bf-sde> 
bf-sde> 
bf-sde> 
bf-sde> 
bf-sde> I20201230 09:48:29.259188    43 p4_service.cc:753] Controller (connection_id: 1, election_id: 1, uri: ipv4:127.0.0.1:59008) is connected as MASTER for node (aka device) with ID 1.
bf_switchd: starting warm init for dev_id 0 mode 1 serdes_upgrade 0 
Skip diag lib deinit
\2020-12-30 09:48:29.517100 BF_BFRT ERROR - getStartBit:2529 ERROR hdr.udp.$valid Context json node not found for the key field. Hence unable to determine the start bit
2020-12-30 09:48:29.517184 BF_SYS ERROR - ASSERTION FAILED: "0" (0) from getStartBit:2530
2020-12-30 09:48:29.517204 BF_BFRT ERROR - keyByteSizeAndOffsetGet:1128 pipe.SwitchEgress.encap.encap_outer_l4 ERROR key field name hdr.udp.$valid not found in match_key_field_name_to_position_map 
2020-12-30 09:48:29.517218 BF_SYS ERROR - ASSERTION FAILED: "0" (0) from keyByteSizeAndOffsetGet:1129
2020-12-30 09:48:29.517231 BF_BFRT ERROR - parseTable:1549 pipe.SwitchEgress.encap.encap_outer_l4 ERROR in processing key field hdr.udp.$valid while trying to get field size and offset
2020-12-30 09:48:29.517252 BF_SYS ERROR - ASSERTION FAILED: "0" (0) from parseTable:1550
2020-12-30 09:48:31.145585 BF_BFRT ERROR - bfrtTableFromIdGet:2363 Table_id "37403834" not found
2020-12-30 09:48:31.145621 BF_BFRT ERROR - bfrtTableFromNameGet:2353 Table "pipe.SwitchEgress.encap.encap_outer_l4" not found
E20201230 09:48:31.145663    45 bfrt_id_mapper.cc:130] StratumErrorSpace::ERR_INTERNAL: Unable to find bfrt ID for P4Info entity pipe.SwitchEgress.encap.encap_outer_l4 with ID 37403834.
E20201230 09:48:31.145674    45 bfrt_id_mapper.cc:38] Return Error: BuildMapping(table.preamble().id(), table.preamble().name(), bfrt_info) failed with StratumErrorSpace::ERR_INTERNAL: Unable to find bfrt ID for P4Info entity pipe.SwitchEgress.encap.encap_outer_l4 with ID 37403834.
E20201230 09:48:31.145678    45 bf_sde_wrapper.cc:1176] Return Error: bfrt_id_mapper_->PushForwardingPipelineConfig(device_config, bfrt_info_) failed with StratumErrorSpace::ERR_INTERNAL: Unable to find bfrt ID for P4Info entity pipe.SwitchEgress.encap.encap_outer_l4 with ID 37403834.
E20201230 09:48:31.145685    45 bfrt_node.cc:91] Return Error: bf_sde_interface_->AddDevice(device_id_, bfrt_config_) failed with StratumErrorSpace::ERR_INTERNAL: Unable to find bfrt ID for P4Info entity pipe.SwitchEgress.encap.encap_outer_l4 with ID 37403834.
E20201230 09:48:31.145689    45 bfrt_switch.cc:74] Return Error: bfrt_node->PushForwardingPipelineConfig(config) failed with StratumErrorSpace::ERR_INTERNAL: Unable to find bfrt ID for P4Info entity pipe.SwitchEgress.encap.encap_outer_l4 with ID 37403834.
E20201230 09:48:31.145709    45 error_buffer.cc:30] (p4_service.cc:451): Failed to set forwarding pipeline config for node 1: Unable to find bfrt ID for P4Info entity pipe.SwitchEgress.encap.encap_outer_l4 with ID 37403834. 
I20201230 09:48:31.146071    38 bf_chassis_manager.cc:879] State of port 1 in node 1 (SDK port 0): DOWN.
I20201230 09:48:31.146111    38 bf_chassis_manager.cc:879] State of port 2 in node 1 (SDK port 1): DOWN.
I20201230 09:48:31.146117    38 bf_chassis_manager.cc:879] State of port 3 in node 1 (SDK port 2): DOWN.
I20201230 09:48:31.146122    38 bf_chassis_manager.cc:879] State of port 4 in node 1 (SDK port 3): DOWN.
I20201230 09:48:31.146128    38 bf_chassis_manager.cc:879] State of port 5 in node 1 (SDK port 4): DOWN.
I20201230 09:48:31.146133    38 bf_chassis_manager.cc:879] State of port 6 in node 1 (SDK port 5): DOWN.
I20201230 09:48:31.146138    38 bf_chassis_manager.cc:879] State of port 7 in node 1 (SDK port 6): DOWN.
I20201230 09:48:31.146142    38 bf_chassis_manager.cc:879] State of port 8 in node 1 (SDK port 7): DOWN.
I20201230 09:48:31.146145    38 bf_chassis_manager.cc:879] State of port 9 in node 1 (SDK port 8): DOWN.
I20201230 09:48:31.146148    38 bf_chassis_manager.cc:879] State of port 10 in node 1 (SDK port 9): DOWN.
I20201230 09:48:31.146152    38 bf_chassis_manager.cc:879] State of port 11 in node 1 (SDK port 10): DOWN.
I20201230 09:48:31.146154    38 bf_chassis_manager.cc:879] State of port 12 in node 1 (SDK port 11): DOWN.
I20201230 09:48:31.146158    38 bf_chassis_manager.cc:879] State of port 13 in node 1 (SDK port 12): DOWN.
I20201230 09:48:31.146162    38 bf_chassis_manager.cc:879] State of port 14 in node 1 (SDK port 13): DOWN.
I20201230 09:48:31.146164    38 bf_chassis_manager.cc:879] State of port 15 in node 1 (SDK port 14): DOWN.
I20201230 09:48:31.146196    38 bf_chassis_manager.cc:879] State of port 16 in node 1 (SDK port 15): DOWN.
I20201230 09:48:31.146205    38 bf_chassis_manager.cc:879] State of port 17 in node 1 (SDK port 16): DOWN.
I20201230 09:48:31.147102    43 p4_service.cc:780] Controller (connection_id: 1, election_id: 1, uri: ipv4:127.0.0.1:59008) which was MASTER for node (aka device) with ID 1 is disconnected. The node is now orphan :(
I20201230 09:57:47.824625    45 p4_service.cc:753] Controller (connection_id: 1, election_id: 1, uri: ipv4:127.0.0.1:59228) is connected as MASTER for node (aka device) with ID 1.
bf_switchd: starting warm init for dev_id 0 mode 1 serdes_upgrade 0 
-2020-12-30 09:57:48.084567 BF_BFRT ERROR - getStartBit:2529 ERROR hdr.udp.$valid Context json node not found for the key field. Hence unable to determine the start bit
2020-12-30 09:57:48.084648 BF_SYS ERROR - ASSERTION FAILED: "0" (0) from getStartBit:2530
2020-12-30 09:57:48.084666 BF_BFRT ERROR - keyByteSizeAndOffsetGet:1128 pipe.SwitchEgress.encap.encap_outer_l4 ERROR key field name hdr.udp.$valid not found in match_key_field_name_to_position_map 
2020-12-30 09:57:48.084680 BF_SYS ERROR - ASSERTION FAILED: "0" (0) from keyByteSizeAndOffsetGet:1129
2020-12-30 09:57:48.084693 BF_BFRT ERROR - parseTable:1549 pipe.SwitchEgress.encap.encap_outer_l4 ERROR in processing key field hdr.udp.$valid while trying to get field size and offset
2020-12-30 09:57:48.084706 BF_SYS ERROR - ASSERTION FAILED: "0" (0) from parseTable:1550
2020-12-30 09:57:48.084902 BF_BFRT  WARN - parseTable:1655 Action SwitchIngress.lag.lag_miss present in context json but not in bfrt json                  for table pipe.SwitchIngress.lag.lag
2020-12-30 09:57:48.084907 BF_BFRT  WARN - parseTable:1655 Action SwitchIngress.lag.set_port present in context json but not in bfrt json                  for table pipe.SwitchIngress.lag.lag
2020-12-30 09:57:48.084966 BF_BFRT  WARN - parseTable:1655 Action NoAction present in context json but not in bfrt json                  for table pipe.SwitchIngress.nexthop.ecmp
2020-12-30 09:57:48.084969 BF_BFRT  WARN - parseTable:1655 Action SwitchIngress.nexthop.set_nexthop_attributes present in context json but not in bfrt json                  for table pipe.SwitchIngress.nexthop.ecmp
2020-12-30 09:57:48.085137 BF_BFRT  WARN - parseTable:1655 Action NoAction present in context json but not in bfrt json                  for table pipe.SwitchIngress.port_mapping.port_vlan_to_bd_mapping
2020-12-30 09:57:48.085141 BF_BFRT  WARN - parseTable:1655 Action SwitchIngress.port_mapping.set_bd_attributes present in context json but not in bfrt json                  for table pipe.SwitchIngress.port_mapping.port_vlan_to_bd_mapping
2020-12-30 09:57:49.491623 BF_BFRT ERROR - bfrtTableFromIdGet:2363 Table_id "37403834" not found
2020-12-30 09:57:49.491657 BF_BFRT ERROR - bfrtTableFromNameGet:2353 Table "pipe.SwitchEgress.encap.encap_outer_l4" not found
E20201230 09:57:49.491703    47 bfrt_id_mapper.cc:130] StratumErrorSpace::ERR_INTERNAL: Unable to find bfrt ID for P4Info entity pipe.SwitchEgress.encap.encap_outer_l4 with ID 37403834.
E20201230 09:57:49.491716    47 bfrt_id_mapper.cc:38] Return Error: BuildMapping(table.preamble().id(), table.preamble().name(), bfrt_info) failed with StratumErrorSpace::ERR_INTERNAL: Unable to find bfrt ID for P4Info entity pipe.SwitchEgress.encap.encap_outer_l4 with ID 37403834.
E20201230 09:57:49.491721    47 bf_sde_wrapper.cc:1176] Return Error: bfrt_id_mapper_->PushForwardingPipelineConfig(device_config, bfrt_info_) failed with StratumErrorSpace::ERR_INTERNAL: Unable to find bfrt ID for P4Info entity pipe.SwitchEgress.encap.encap_outer_l4 with ID 37403834.
E20201230 09:57:49.491727    47 bfrt_node.cc:91] Return Error: bf_sde_interface_->AddDevice(device_id_, bfrt_config_) failed with StratumErrorSpace::ERR_INTERNAL: Unable to find bfrt ID for P4Info entity pipe.SwitchEgress.encap.encap_outer_l4 with ID 37403834.
E20201230 09:57:49.491732    47 bfrt_switch.cc:74] Return Error: bfrt_node->PushForwardingPipelineConfig(config) failed with StratumErrorSpace::ERR_INTERNAL: Unable to find bfrt ID for P4Info entity pipe.SwitchEgress.encap.encap_outer_l4 with ID 37403834.
E20201230 09:57:49.491748    47 error_buffer.cc:30] (p4_service.cc:451): Failed to set forwarding pipeline config for node 1: Unable to find bfrt ID for P4Info entity pipe.SwitchEgress.encap.encap_outer_l4 with ID 37403834. 
I20201230 09:57:49.491750    38 bf_chassis_manager.cc:879] State of port 1 in node 1 (SDK port 0): DOWN.
I20201230 09:57:49.491776    38 bf_chassis_manager.cc:879] State of port 2 in node 1 (SDK port 1): DOWN.
I20201230 09:57:49.491784    38 bf_chassis_manager.cc:879] State of port 3 in node 1 (SDK port 2): DOWN.
I20201230 09:57:49.491787    38 bf_chassis_manager.cc:879] State of port 4 in node 1 (SDK port 3): DOWN.
I20201230 09:57:49.491791    38 bf_chassis_manager.cc:879] State of port 5 in node 1 (SDK port 4): DOWN.
I20201230 09:57:49.491796    38 bf_chassis_manager.cc:879] State of port 6 in node 1 (SDK port 5): DOWN.
I20201230 09:57:49.491801    38 bf_chassis_manager.cc:879] State of port 7 in node 1 (SDK port 6): DOWN.
I20201230 09:57:49.491804    38 bf_chassis_manager.cc:879] State of port 8 in node 1 (SDK port 7): DOWN.
I20201230 09:57:49.491807    38 bf_chassis_manager.cc:879] State of port 9 in node 1 (SDK port 8): DOWN.
I20201230 09:57:49.491812    38 bf_chassis_manager.cc:879] State of port 10 in node 1 (SDK port 9): DOWN.
I20201230 09:57:49.491820    38 bf_chassis_manager.cc:879] State of port 11 in node 1 (SDK port 10): DOWN.
I20201230 09:57:49.491827    38 bf_chassis_manager.cc:879] State of port 12 in node 1 (SDK port 11): DOWN.
I20201230 09:57:49.491832    38 bf_chassis_manager.cc:879] State of port 13 in node 1 (SDK port 12): DOWN.
I20201230 09:57:49.491837    38 bf_chassis_manager.cc:879] State of port 14 in node 1 (SDK port 13): DOWN.
I20201230 09:57:49.491840    38 bf_chassis_manager.cc:879] State of port 15 in node 1 (SDK port 14): DOWN.
I20201230 09:57:49.491845    38 bf_chassis_manager.cc:879] State of port 16 in node 1 (SDK port 15): DOWN.
I20201230 09:57:49.491852    38 bf_chassis_manager.cc:879] State of port 17 in node 1 (SDK port 16): DOWN.
I20201230 09:57:49.492836    45 p4_service.cc:780] Controller (connection_id: 1, election_id: 1, uri: ipv4:127.0.0.1:59228) which was MASTER for node (aka device) with ID 1 is disconnected. The node is now orphan :(
  ```
</details>

**Comment by davidgengenbach** [comment]
Got it working. An environment variable was not exported correctly which started Stratum with the 9.2.0 BF-SDE instead of 9.3.0.
Just for completeness sake:

```shell
git clone --branch 2020-12-18 https://github.com/stratum/stratum.git
cd stratum

export SDE_TAR="..." # See README
# Only once per SDE version
sudo -E ./stratum/hal/bin/barefoot/docker/build-tofino-model-container.sh $SDE_TAR

# Run in background. Logs can be seen with sudo docker logs --follow tofino-model
sudo -E docker run --detach --name tofino-model --rm -it --privileged --network=host stratumproject/tofino-model:9.3.0
# Wait a little for the tofino-model to be initialized
sleep 3
sudo -E DOCKER_IMAGE="stratumproject/stratum-bfrt" DOCKER_IMAGE_TAG="9.3.0" PLATFORM="barefoot-tofino-model" stratum/hal/bin/barefoot/docker/start-stratum-container.sh -bf_sim  -bf_switchd_background=false  -enable_onlp=false
```
:warning: Important: convert the config as described in the README!

**Comment by pudelkoM** [comment]
Which variable was causing this and do you think documentation should be improved?

**Comment by davidgengenbach** [comment]
`SDE_VERSION="9.3.0"` (or `DOCKER_IMAGE_TAG="9.3.0"`)  and `DOCKER_IMAGE="stratumproject/stratum-bfrt"` when starting Stratum via  `stratum/hal/bin/barefoot/docker/start-stratum-container.sh`.

In the [README](https://github.com/stratum/stratum/blob/master/stratum/hal/bin/barefoot/README.run.md#start-tofino-model) there is the mention of the SDE version for the `tofino-model` image - but not for Stratum!
If I remember correctly, I had to look for the environment variables in the `start-stratum-container.sh` myself - I do not recall reading about it in a README.
I think it would help to add a small hint regarding that - I am not sure whether the `stratum-bfrt` image is needed but I think the TNA program I had did not work otherwise (but I am not sure about that one).

**Comment by davidgengenbach** [comment]
Just to be precise: the problem with the un-exported environment variable was not due to Stratum - it was my fault. I had a typo and did not realize that the `start-stratum-container.sh` script did not pick my env variable up because of that.

**Comment by pudelkoM** [comment]
Have you seen this section https://github.com/stratum/stratum/blob/master/stratum/hal/bin/barefoot/README.run.md#environment-variables-for-start-stratum-containersh ?
It's a bit newer.

**Comment by davidgengenbach** [comment]
@pudelkoM I oversaw this paragraph as I looked at the script itself. It might have been nice to have a comment that both `stratumproject/stratum-bf` and `stratumproject/stratum-bfrt` are valid options for `DOCKER_IMAGE` since I only got my TNA 
config pushed with the `bfrt` version if I remember correctly.

That you can use `stratum-bfrt` is mentioned in the last part of the README but not how to achieve it - namely `DOCKER_IMAGE="stratumproject/stratum-bfrt"`.
